### ISSUE TITLE
Default Variables for Sass Mixins

### ISSUE THREAD

**Issue by ayuhito** [fontsource/fontsource#129]
Continuing on from #64. Suggestion and quoted content attributed to @cbirdsong.

The proposal suggests that the mixins should include default variables to reduce repetitive code and become more manageable.
e.g. `$fontsourceDisplay: swap !default;` or `$fontsourceWeight: 400 !default;`.

> For defaults, relying on the mixin arguments is fine for things that are likely to change each time you call the mixin, but it seems really unlikely you'd change font-display, file path and character set for each individual typeface. That feels like an advanced feature and it's good the mixin allows for it, but it introduces a lot of repetition.
> 
> I imagine the far more common scenario is someone deciding "we're using font-display: optional; and keeping our compiled files in /dist/", and those people have to repeat those instructions to each "@include fontFace` over and over again. Compare these two code blocks:

```
@use "@fontsource/open-sans/scss/mixins" as OpenSans;
@use "@fontsource/barlow-condensed/scss/mixins" as Barlow;

@include OpenSans.fontFace($fontDir: "/dist/node_modules/@fontsource/open-sans/files", $display: optional);
@include OpenSans.fontFace($fontDir: "/dist/node_modules/@fontsource/open-sans/files", $display: optional, $style: italic);

@include OpenSans.fontFace($fontDir: "/dist/node_modules/@fontsource/open-sans/files", $display: optional, $weight: 700);
@include OpenSans.fontFace($fontDir: "/dist/node_modules/@fontsource/open-sans/files",  $display: optional, $weight: 700,  $style: "italic");

@include Barlow.fontFace($fontDir: "/dist/node_modules/@fontsource/barlow-condensed/files", $display: optional, $weight: 600);
@include Barlow.fontFace($fontDir: "/dist/node_modules/@fontsource/barlow-condensed/files",  $display: optional, $weight: 600,  $style: "italic");
```
```
@use "@fontsource/open-sans/scss/mixins" as OpenSans;
@use "@fontsource/barlow-condensed/scss/mixins" as Barlow;

$fontsourceDir: "/dist/node_modules/@fontsource/"
$fontsourceDisplay: optional;

@include OpenSans.fontFace($weight: 400, $style: normal);
@include OpenSans.fontFace($weight: 400, $style: italic);

@include OpenSans.fontFace($weight: 700, $style: normal);
@include OpenSans.fontFace($weight: 700, $style: "italic");

@include Barlow.fontFace($weight: 600, $style: normal);
@include Barlow.fontFace($weight: 600,  $style: "italic");
```

> The second way leads to much more maintainable code than the first. You can understand what typefaces are being loaded at a glance*, adding another weight is a simple copy/paste, and modifying the file path requires editing only a single line.

**Comment by ayuhito** [comment]
Released in 4.2.0.

**Comment by davidjb** [comment]
@DecliningLotus Just opened https://github.com/fontsource/fontsource/issues/153 - a few suggestions there on how to fix the problem introduced by these changes
